hive-metastore/src/test/java/org/apache/iceberg/hive/TestClientPool.java

```diff
@@ -51,4 +81,918 @@ protected void close(Object client) {
 
     }
   }
+
+  private static class MockHiveClientPool extends ClientPool<IMetaStoreClient, Exception> {
```

Comment:
Shall we just extend the HiveClientPool here? Or we want to test the ClientPool too?